### PR TITLE
Feat/srp metrics

### PR DESCRIPTION
**Pull request by montelaidev**
## Explanation

Adds metrics for SRP reveal. 

See: https://github.com/MetaMask/MetaMask-planning/issues/268

## Manual Testing Steps

1. Click on Security & privacy tab and clicking Reveal Secret Recovery Phrase  a `Clicked Reveal Secret Recovery` event will be fired.
2. When arriving at Secret Recovery Phrase page, `Reveal Secret Recovery Phrase Page` event will be fired.
- If Cancel is clicked, `Reveal SRP Cancelled` event will be fired
3. When clicking the Hold To Reveal SRP button, `Reveal SRP Click Started` event will be fired, and after hold completion, `Reveal SRP Completed` event is fired and `Views SRP` is fired.
4. If the user clicks the QR tab, `Views SRP QR Code` event is fired
5. If the user clicks Copy to Clipboard, `Copies SRP to clipboard` event is fired.
6. When the user is done and clicks Close, `Clicks Done with SRP` is fired

## Pre-merge author checklist

- [x ] I've clearly explained:
  - [x ] What problem this PR is solving
  - [ x] How this problem was solved
  - [ x] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: ui/components/app/modals/hold-to-reveal-modal/hold-to-reveal-modal.js [hunk 1]
```diff
@@ -20,13 +20,21 @@ import {
...
 
-const HoldToRevealModal = ({ onLongPressed, hideModal }) => {
+const HoldToRevealModal = ({ onLongPressed, willHide, hideModal }) => {
   const t = useI18nContext();
+  const trackEvent = useContext(MetaMetricsContext);
 
   const unlock = () => {
     onLongPressed();
-    hideModal();
+    if (willHide) {
```
[the remark this answers is not in the document]
Comment (montelaidev) [reply ui/components/app/modals/hold-to-reveal-modal/hold-to-reveal-modal.js]: its another feature. i'll move this into another pr

File: ui/components/app/modals/hold-to-reveal-modal/hold-to-reveal-modal.test.js [hunk 2]
```diff
@@ -12,23 +12,25 @@ import {
   holdToRevealContent5,
   holdToRevealTitle,
 } from '../../../../../app/_locales/en/messages.json';
+import {
+  EVENT,
+  EVENT_NAMES,
+} from '../../../../../shared/constants/metametrics';
+// import MockProvider from './test-helper/mockProvider';
```
Comment (darkwing) [inline ui/components/app/modals/hold-to-reveal-modal/hold-to-reveal-modal.test.js:19]: Comment needs removal.

File: ui/pages/keychains/reveal-seed.js [hunk 3]
```diff
@@ -144,9 +144,39 @@ const RevealSeedPage = () => {
   };
 
   const renderRevealSeedContent = () => {
+    // default fir SRP_VIEW_SRP_TEXT event because this is the first thing showed after rendering
```
Comment (darkwing) [inline ui/pages/keychains/reveal-seed.js:147]: Tiny verbiage nits:

* I think `fir` is meant to be `for`
* `showed` can be `shown`

**Review by darkwing — CHANGES_REQUESTED** [review]
Looks great overall, just needs a few tiny updates!

**Review by darkwing — DISMISSED** [review]

**Review by HowardBraham — DISMISSED** [review]

File: ui/components/app/modals/hold-to-reveal-modal/hold-to-reveal-modal.js [hunk 4]
```diff
@@ -53,7 +60,16 @@ const HoldToRevealModal = ({ onLongPressed, hideModal }) => {
...
-          onClick={handleCancel}
+          onClick={() => {
+            trackEvent({
+              category: MetaMetricsEventCategory.Keys,
+              event: MetaMetricsEventName.SrpHoldToRevealCloseClicked,
+              properties: {
+                key_type: MetaMetricsEventKeyType.Srp,
+              },
+            });
+            handleCancel();
```
Comment (HowardBraham) [inline ui/components/app/modals/hold-to-reveal-modal/hold-to-reveal-modal.js:71]: I wonder if this `trackEvent` should go inside `handleCancel`

**Review by HowardBraham — DISMISSED** [review]

File: ui/components/app/hold-to-reveal-button/hold-to-reveal-button.js [hunk 5]
```diff
@@ -93,6 +121,7 @@ export default function HoldToRevealButton({ buttonText, onLongPressed }) {
         <Box className="hold-to-reveal-button__absolute-fill">
           <svg className="hold-to-reveal-button__circle-svg">
             <circle
+              aria-label="circle-locked"
```
Comment (darkwing) [inline ui/components/app/hold-to-reveal-button/hold-to-reveal-button.js:124]: Are these meant to be in this PR?  They don't seem related to metrics.

**Review by darkwing — COMMENTED** [review]
Just one comment, but overall looks good.

File: ui/components/app/hold-to-reveal-button/hold-to-reveal-button.js (hunk 5 above)
Comment (montelaidev) [reply ui/components/app/hold-to-reveal-button/hold-to-reveal-button.js:124]: yes, these are for the tests to trigger the unlock

**Review by HowardBraham — APPROVED** [review]

**Review by darkwing — APPROVED** [review]
